Userland/Libraries/LibWeb/CSS/StyleComputer.cpp

```diff
@@ -1430,86 +1422,94 @@ bool PropertyDependencyNode::has_cycles()
 
 void StyleComputer::build_rule_cache_if_needed() const
 {
-    if (m_rule_cache)
+    if (m_author_rule_cache && m_user_agent_rule_cache)
         return;
     const_cast<StyleComputer&>(*this).build_rule_cache();
 }
 
 void StyleComputer::build_rule_cache()
 {
-    // FIXME: Make a rule cache for UA style as well.
+    auto make_rule_cache = [&](CascadeOrigin cascade_origin) -> NonnullOwnPtr<RuleCache> {
```

Comment:
I feel like this 68-line lambda that only captures `this` would have a better home as its own member function

Reply:
You have a point!